radis/lbl/base.py

```diff
@@ -2035,9 +2040,21 @@ def _calc_Q(molecule, iso, state):
                         Qref, Qgas = _calc_Q(molecule, iso, self.input.state)
                         Qref_arr[iso] = Qref
                         Qgas_arr[iso] = Qgas
-                # ... the trick below is that iso is used as index in the array
-                df1["Qref"] = Qref_arr.take(df1.iso)
-                df1["Qgas"] = Qgas_arr.take(df1.iso)
+
+                # convert dictionaries to lists
+                Qref_dict = list_to_dict(Qref_arr)
+                Qgas_dict = list_to_dict(Qgas_arr)
+
+                Qref_Qgas_ratio = {
+                    k: Qref_dict.get(k, 0) / Qgas_dict.get(k, 0)
+                    for k in set(Qref_dict) | set(Qgas_dict)
+                }
+
+                # copy iso column
+                df1["S"] = df1["iso"]
+
+                # map the ratios in the dictionary to the iso column vallues
+                df1["S"].map(Qref_Qgas_ratio)
```

Comment:
cf Slack comment : 

you should be able to simplify a lot all of these lines. 
Eventually there could be only something like :    L2077:

```
line_strength = df1.int * df1["iso"].map(Qref_Qgas_ratio] 
```
Alternative if only one isotope : 
```
line_strength = df1.int * Qref / Qgas
```

Reply:
```
line_strength = df1.int * df1["iso"].map(Qref_Qgas_ratio] 
```

I don't think this is the right syntax we can use because if I do something like

```
df1["new"] = df1["iso"].map(Qref_Qgas_ratio)
```

It will give me `NaN` in all the entries of the new `new` column and hence prefilling `df["S]` with the values in `iso`. But other than that I have simplified the code by removing the redundant arrays.